Refine semantic conventions stability (#1636)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/versioning.md

## Major version bump

There are a few cases when a major bump may occur.

* Breaking change of a stable functionality or configuration.
* Removal of a stable functionality or configuration.

> **Exception:** Changes connected with instrumentation libraries are handled differently.
> There are handled the same way as experimental functionality and configuration.

Reviewer: ```suggestion
**Exception:** Changes related to instrumentation libraries are handled
 the same way as experimental functionality and configuration changes.
```
